src/transformers/trainer.py

```diff
@@ -340,14 +340,16 @@ def num_examples(self, dataloader: DataLoader) -> int:
         """
         return len(dataloader.dataset)
 
-    def train(self, model_path: Optional[str] = None):
+    def train(self, model_path: Optional[str] = None, close_tb_writer: Optional[bool] = True):
         """
         Main training entry point.
 
         Args:
             model_path:
                 (Optional) Local path to model if model to train has been instantiated from a local path
                 If present, we will try reloading the optimizer/scheduler states from there.
+            close_tb_writer:
+                (Optional) Whether or not to close tb_writer after training is done.
```

Comment:
I didn't get why we need this option until I read the issue https://github.com/huggingface/transformers/issues/5329 so you may want to add more explanation here.

Reply:
maybe add some text "to be able to write more data to TensorBoard" or equivalent

Also, I'd rather have the option in the `__init__`.

Otherwise, the feature looks reasonable to me, thanks for adding it